docs/proposals/Annotation-Based-Artifact-Filtration.md

```diff
@@ -0,0 +1,131 @@
+# Annotation Based Artifact Filteration
+
+## Problem/Motivation
+
+When configuring a verifier in Ratify, we set the artifact type the verifier should work on. In such case, Ratify will verify all referrers of a given subject that have a matching artifact type using the verifier.
+In some cases, this could lead to a wrong behavior. For instance, Vulnerability Artifacts are outdated once a new artifact is written to the repository, as such there is no use for verifying both the new one and the old one. On the other hand, in performing signature verification using Notary, we may wish to attest more than one signature, because there may be multiple signatures attached to a given artifact, but our "tenant" only trusts a subset of them. 
+
+The issue with verifying all the matching artifacts could also lead to performance issues, each "verification" process hides within a request to pull the artifcat manifest, and the blobs containing the actual data.
```

Comment:
The verifier api includes a `CanVerify` which is invoked by Ratify's executor prior to even invoking the `Verify` method. If the verifier defines it's own behavior for annotation filtering here, then there would be no extra registry request overhead nor executor-level logic required.

Reply:
^^  I think this is a good point.

Reply:
@akashsinghal, I agree, having the annotation based filtering in the CanVerify method would offer a clean solution for that part of the proposal.
We could also define how to setup the filtering in a descriptive way in the Verifier configuration to enable external plugin to make use of this. Since the CanVerify method is implemented internally in the VerifierPlugin "verifier" in Ratify.

As for age based, I don't see a way without first pulling all the references and sorting them/placing them in some sort of heap to enable fast access to the latest artifact.